lens.cabal

```diff
@@ -406,7 +408,8 @@ test-suite hunit
   hs-source-dirs: tests
   default-language: Haskell2010
 
-  if !flag(test-hunit)
+  -- test-framework-th doesn't work due regex-posix
```

Comment:
Doesn't work on GHCJS in particular, you mean? Can you elaborate more on this?

Reply:
regex-posix uses C code, which doesn’t seem to be shimmed by GHCJS. In general regex-tdfa is better as pure-Haskell lib

Reply:
Ah, I see. How does GHCJS normally handle packages with C code? Is there a repository of GHCJS-specific patches that people use? I'm aware of [`hackage-ghcjs-overlay`](https://github.com/nomeata/hackage-ghcjs-overlay), although that doesn't appear to have any patches at the moment, so surely that can't be the answer I'm seeking.

Reply:
Something is built-in GHCJS itself: https://github.com/ghcjs/ghcjs/tree/ghc-8.6/lib/patches

Reply:
I see. If I'm looking at that right, all of those are boot libraries, correct? What happens if someone wishes to patch something that's not a boot library?

Reply:
That's very good question. I think `hackage-ghcjs-overlay` existed for that kind of reason.

But now I'd just submit patches to the libraries. In this particular case, change `test-framework-th` to use `regex-tdfa`. Yet if one goes as far, maybe switching to `tasty` is an option too.

Reply:
To be clear, I'm not super bothered by disabling the test suite for GHCJS in this particular situation. In the event that testing against GHCJS becomes more popular, however, it might be worthwhile to investigate this further. I'm content for now knowing that there's at least a glimmer of hope in `hackage-ghcjs-overlay`.

Reply:
Found: https://github.com/ghcjs/ghcjs/tree/ghc-8.6/lib/boot/shims there are shims